### ISSUE TITLE
Support for comment nodes

### ISSUE THREAD

**Issue by r24y** [snabbdom/snabbdom#142]
I'm interested in using snabbdom server-side, but as far as I can tell it lacks the ability to handle comments. This is useful for (among other things) adding browser-conditional comments:

```
<!--[if lte IE 8]><script src="..."></script><!endif-->
```

Other than emitting a `<div>` with `innerHTML` set for a comment, how can I add comments?

**Comment by paldepind** [comment]
Are you using snabbdom-to-html?

I don't think there currently is a way to achieve what you want.

How whould you like for this to work ideally?

**Comment by katyo** [comment]
I have same issue also. In any case we must provide support for comment nodes in html because the DOM supports it too (see [document.createComment()]).

[document.CreateComment()]: https://www.w3.org/TR/REC-DOM-Level-1/level-one-core.html#method-createComment

**Comment by r24y** [comment]
@katyo great point! I'm changing this issue to be a bit more general; once comment nodes are supported then browser conditionals should "just work" by putting text into comment nodes.

```diff
  Issue #142
- `if lte IE 8` and similar comments
+  Support for comment nodes
```

**Comment by r24y** [comment]
I'm not picky on the API, but for the sake of giving a strawman:

```javascript
h('!', ['comment content'])
```

Of course this may end up needing all sorts of special cases; comments don't have attributes, a user might try to put a hook on a comment node, etc. It would be pretty easy to implement comment nodes, but tough to guard against the ways people can break them in the real world!

**Comment by paldepind** [comment]
I think your proposed API looks very fine :+1: 

I am not too concerned that users can in theory put attributes, hooks, etc. on comment nodes.

**Comment by paldepind** [comment]
This was closed by #229.
